### PR TITLE
Fix some bugs in consteval related to different type combinations.

### PR DESCRIPTION
**Pull request by stellaraccident**
With these fixes, the test suite builds successfully (with constant hoisting enabled).

### CODE REVIEW

File: compiler/src/iree/compiler/ConstEval/Runtime.cpp [hunk 1]
```diff
@@ -260,6 +308,52 @@ LogicalResult FunctionCall::addArgument(Location loc, Attribute attr) {
...
+    switch (APFloat::SemanticsToEnum(apValue.getSemantics())) {
+    // case APFloat::S_IEEEhalf:
+    // case APFloat::S_BFloat:
+    case APFloat::S_IEEEsingle:
+      value = iree_vm_value_make_f32(apValue.convertToFloat());
+      break;
+    case APFloat::S_IEEEdouble:
+      value = iree_vm_value_make_f64(apValue.convertToDouble());
+      break;
+    default:
```
Comment (benvanik) [inline compiler/src/iree/compiler/ConstEval/Runtime.cpp]: useful to call out unhandled things but can get compiler error/lsc renaming niceness by having them in code:
```suggestion
    case APFloat::S_IEEEsingle:
      value = iree_vm_value_make_f32(apValue.convertToFloat());
      break;
    case APFloat::S_IEEEdouble:
      value = iree_vm_value_make_f64(apValue.convertToDouble());
      break;
    default:
    case APFloat::S_IEEEhalf:
    case APFloat::S_BFloat:
```

File: compiler/src/iree/compiler/ConstEval/Runtime.cpp [hunk 2]
```diff
@@ -118,20 +121,14 @@ static TypedAttr createAttributeFromRawData(Location loc,
...
-      llvm::SmallVector<bool> boolVector(rawBuffer.begin(), rawBuffer.end());
-      ArrayRef<bool> boolArray(boolVector.data(), boolVector.size());
-      return DenseElementsAttr::get(tensorType, boolArray);
-    }
+    // Note: cannot use std::vector because it specializes bool in a way
+    // that is not compatible with ArrayRef.
+    llvm::SmallVector<bool> boolVector(rawBuffer.begin(), rawBuffer.end());
+    ArrayRef<bool> boolArray(boolVector.data(), boolVector.size());
+    return DenseElementsAttr::get(tensorType, boolArray);
+    // }
```
Comment (benvanik) [inline compiler/src/iree/compiler/ConstEval/Runtime.cpp]: unintended?

**Review by benvanik — APPROVED** [review]

File: compiler/src/iree/compiler/ConstEval/Runtime.cpp (hunk 1 above)
Comment (stellaraccident) [reply compiler/src/iree/compiler/ConstEval/Runtime.cpp]: Left a long-hand comment instead: there is a long list of semantics types that we will not support at this level.

File: compiler/src/iree/compiler/ConstEval/Runtime.cpp (hunk 2 above)
Comment (stellaraccident) [reply compiler/src/iree/compiler/ConstEval/Runtime.cpp]: Indeed.

**Pull request merged**